python/gLaSDI/examples/1DBurgerEqn/generate_data.ipynb

```diff
@@ -0,0 +1,275 @@
+    "        data.append(tmp)\n",
+    "        param.append(np.array([i, j]))\n",
+    "data_all = {}\n",
+    "data_all['data'] = data\n",
+    "data_all['param'] = param\n",
+    "    \n",
+    "\n",
+    "num_case = len(data_all['data'])\n",
+    "param = data_all['param']\n",
+    "time1 = 0\n",
+    "time2 = 0\n",
+    "time3 = 0\n",
+    "print(f'number of cases: {num_case}')\n",
+    "for i in range(num_case):\n",
+    "    print(f\"Case {i+1}: A: {param[i][0]:.2f}, W: {param[i][1]:.2f}, t: {data_all['data'][i]['t'].shape}, x: {data_all['data'][i]['x'].shape}, dx: {data_all['data'][i]['dx'].shape}\")\n",
+    "    time1 += data_all['data'][i]['time_fom'][0]\n",
+    "    time2 += data_all['data'][i]['time_fom'][1]\n",
+    "    time3 = time1 + time2\n",
+    "    \n",
+    "print(f\"time for computing x: {time1/num_case:.2f} s\")\n",
+    "print(f\"time for computing dx: {time2/num_case:.2f} s\")\n",
+    "print(f'total time: {time3/num_case:.2f} s')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "94264eb2-c7fe-480e-9734-690bb83188ca",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "t = np.linspace(0,tstop,nt+1)\n",
+    "x = np.linspace(x0,x1,nx)\n",
+    "step_list = np.linspace(0,nt,4,dtype=int)\n",
+    "\n",
+    "fig1 = plt.figure(figsize=(9,6))\n",
+    "idx = np.arange(0,t.size,50)\n",
+    "ax = fig1.add_subplot(111)\n",
+    "for i,step in enumerate(step_list):\n",
+    "    ax.plot(x, data_all['data'][0]['x'][step,:], '-', lw=2, c=cmap(i), label=f't={t[step]:.1f} s')\n",
+    "\n",
+    "ax.set_xlabel('x')\n",
+    "ax.set_ylabel('u')\n",
+    "ax.set_xlim(x.min(),x.max())\n",
+    "ax.set_ylim(bottom=0)\n",
+    "ax.tick_params(axis='both', labelsize=24)\n",
+    "ax.legend(loc='upper left', frameon=False, fontsize=18)\n",
+    "plt.tight_layout()\n",
+    "# plt.savefig(f\"./1Dburger_physical_dynamics.png\",bbox_inches='tight')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9353d185",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# save data\n",
+    "if save_data:\n",
+    "    data_path = '/g/g92/he10/Research/data/1DBurgerEqn'\n",
+    "    if not os.path.exists(data_path):\n",
+    "        os.makedirs(data_path)\n",
+    "    if num_case > 1:\n",
+    "        pickle.dump(data_all, open(data_path + f\"/local{num_case}.p\", \"wb\"))\n",
+    "    else:\n",
+    "        pickle.dump(data_all, open(data_path + f\"/local{num_case}_A{amp[0]:.2f}_W{width[0]:.2f}.p\", \"wb\"))"
+   ]
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "tfvenv",
+   "language": "python",
+   "name": "tfvenv"
+  },
+  "language_info": {
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.7.10"
```

Comment:
@dylan-copeland @chldkdtn @xiaolong7 I am not sure how I feel about putting Jupyter Notebook files in the repo. There is a lot of metadata that changes, such as the lines above, whenever someone runs the code. This makes it hard to version control.

https://nextjournal.com/schmudde/how-to-version-control-jupyter

Is there a preference for Jupyter Notebook vs. pure Python files (using something like nbconvert) within libROM?

It seems like the main benefit of Jupyter Notebook is that it would allow us to store completed results within the file, so people could easily visualize them without running any code. But it looks like the output isn't stored here.

Reply:
Sorry for getting back to this so late. I agree with you. The preference must be pure Python files. @xiaolong7 do you think you can change Julyter Notebook to python scripts in this repo?

Reply:
Yes, sure. I will convert the Jupyter notebook to Python scripts in this repo.